### PR TITLE
[BitDefender] New integration

### PR DESCRIPTION
**Pull request by colin-stubbs**
Add initial BitDefender integration release

## What does this PR do?

PR addressing "Add BitDefender integration" [issue #5122](https://github.com/elastic/integrations/issues/5122)

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [X] Linting/Testing/etc

## How to test this PR locally

Use elastic-package test, sample logs and full system tests exist

Alternatively, signup for a BitDefender GravityZone account, which is not free.

## Related issues

- Closes [issue #5122](https://github.com/elastic/integrations/issues/5122)

## Screenshots

Configuration 1

Configuration 2

Dashboard 1

Dashboard 2

### CODE REVIEW

File: packages/bitdefender/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            show_user: true
+            required: true
+            default: https://cloud.gravityzone.bitdefender.com/api/v1.0/jsonrpc/push
+          - name: api_key
+            type: text
+            title: BitDefender GravityZone API Key
+            show_user: true
+            required: true
+owner:
+  github: elastic/integrations
```
Comment (efd6) [inline packages/bitdefender/manifest.yml:76]:
```suggestion
  github: elastic/security-external-integrations
```

**Comment by kalramani** [comment]
@colin-stubbs - are you happy to accept and commit Dan`s change.

**Comment by colin-stubbs** [comment]
@efd6 - missed that last commit suggestion, that's in now.

File: packages/bitdefender/data_stream/push_notifications/_dev/test/pipeline/test-common-config.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    - preserve_duplicate_custom_fields
+  _tmp:
+    bitdefender_id: test1234
+    tenants:
+      "623c18fb12fb8700396d6375": "test_events.tld"
+      "123456789c0318a123456789": "tenant_a.tld"
+      "12345678904ad38123456789": "tenant_b.tld"
+      "12345678904ad33123456789": "tenant_c.tld"
+      "123456789357480123456789": "tenant_d.tld"
+      "123456789a8a356123456789": "tenant_e.tld"
```
[the remark this answers is not in the document]
Comment (colin-stubbs) [reply packages/bitdefender/data_stream/push_notifications/_dev/test/pipeline/test-common-config.yml:13]: 623c18fb12fb8700396d6375 is the company ID that BitDefender uses when they generate a test event internally.

The others are dummy test ID's to simulate having multiple companies as sub-companies/tenants under a master MSSP company/tenant.

File: packages/bitdefender/data_stream/push_notifications/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  fields:
+    - name: id
+      type: keyword
+    - name: jsonrpc
+      type: keyword
+    - name: event
+      type: group
+      description: Fields for Push Notification events
+      fields:
+        - name: _testEvent_
```
[the remark this answers is not in the document]
Comment (colin-stubbs) [reply packages/bitdefender/data_stream/push_notifications/fields/fields.yml:13]: As in a rename ingest pipeline processor?

File: packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -0,0 +1,1255 @@
...
+    copy_from: bitdefender.event.malwareName
+    field: threat.software.name
+    ignore_empty_value: true
+    if: ctx.threat?.software?.name == null
+
+# malware_status
+# malware_type
+# malware.malwareType
+# malware.actionTaken
+# malware.infectedObject
```
[the remark this answers is not in the document]
Comment (colin-stubbs) [reply packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml:970]: Kind of... they're fields that I identified as having potential for mapping to ECS fields in some way, or using for translation to set an ECS field, but I could not, at the time, work out what to do with them.

They can be stripped out if necessary.

File: packages/bitdefender/data_stream/push_notifications/fields/fields.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (efd6) [reply packages/bitdefender/data_stream/push_notifications/fields/fields.yml:13]: Yeah. There's a bunch more that don't match ECS naming conventions, but this is the only one that I think really needs to be changed.

File: packages/bitdefender/data_stream/push_notifications/_dev/test/pipeline/test-common-config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (efd6) [reply packages/bitdefender/data_stream/push_notifications/_dev/test/pipeline/test-common-config.yml:13]: Thanks.

File: packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (efd6) [reply packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml:970]: Can you mark them with a TODO?

**Comment by jamiehynds** [comment]
@colin-stubbs am I correct in saying this PR is for BitDefender GravityZone? If so, the package title/description will need to be adjust to ensure we include vendor + product name. Thanks!

**Comment by colin-stubbs** [comment]
> @colin-stubbs am I correct in saying this PR is for BitDefender GravityZone? If so, the package title/description will need to be adjust to ensure we include vendor + product name. Thanks!

Well it's BitDefender more generally, GravityZone is simply their business product set. Their SOHO products have no SIEM integration capability and likely never will.

File: packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (colin-stubbs) [reply packages/bitdefender/data_stream/push_notifications/elasticsearch/ingest_pipeline/default.yml:970]: Sure...

**Comment by jamiehynds** [comment]
@efd6 @colin-stubbs are there any outstanding items to address within the PR or can we proceed to merging?

**Comment by efd6** [comment]
Just needs to have `elastic-package build` re-run to fix up the docs.

**Comment by jamiehynds** [comment]
> Just needs to have elastic-package build re-run to fix up the docs.

@colin-stubbs is this something you can do to wrap up the PR, or would you prefer us to? Would be great to get this integration out there. Thanks again for the contribution!

**Comment by kalramani** [comment]
@jamiehynds Hi Jamie, Collin is in on long leaves. Can you please do this step and merge this code please.

**Comment by jamiehynds** [comment]
@efd6 given that Colin is on leave for some time, do you think we could wrap up this PR on our side and merge?

**Comment by efd6** [comment]
@kalramani I am not able to push to your branch to move this forward. Would you either make that possible for me or apply and push the attached patches.

[0001-build-docs.patch](https://github.com/elastic/integrations/files/11268486/0001-build-docs.patch)
[0002-fix-append-processor.patch](https://github.com/elastic/integrations/files/11268487/0002-fix-append-processor.patch)
[0003-add-TODOs.patch](https://github.com/elastic/integrations/files/11268488/0003-add-TODOs.patch)

**Comment by CallumDowling** [comment]
> @kalramani I am not able to push to your branch to move this forward. Would you either make that possible for me or apply and push the attached patches.
> 
> [0001-build-docs.patch](https://github.com/elastic/integrations/files/11268486/0001-build-docs.patch) [0002-fix-append-processor.patch](https://github.com/elastic/integrations/files/11268487/0002-fix-append-processor.patch) [0003-add-TODOs.patch](https://github.com/elastic/integrations/files/11268488/0003-add-TODOs.patch)

Hi @efd6, we (@kalramani and I) also don't have access, so options are:
1. wait for Colin
2. clone his read only fork and new PR
?

**Comment by efd6** [comment]
@CallumDowling Would you mind sending a new PR with clone of his fork?

**Comment by kalramani** [comment]
https://github.com/elastic/integrations/pull/5944

**Comment by colin-stubbs** [comment]
> > @kalramani I am not able to push to your branch to move this forward. Would you either make that possible for me or apply and push the attached patches.
> > [0001-build-docs.patch](https://github.com/elastic/integrations/files/11268486/0001-build-docs.patch) [0002-fix-append-processor.patch](https://github.com/elastic/integrations/files/11268487/0002-fix-append-processor.patch) [0003-add-TODOs.patch](https://github.com/elastic/integrations/files/11268488/0003-add-TODOs.patch)
> 
> Hi @efd6, we (@kalramani and I) also don't have access, so options are:
> 
> 1. wait for Colin
> 2. clone his read only fork and new PR
>    ?

Ya'll just need write on the fork? Happy to provide.

**Comment by jamiehynds** [comment]
Hey @colin-stubbs - we managed to resolve it via this PR: https://github.com/elastic/integrations/pull/5944

Integration is now available - https://docs.elastic.co/integrations/bitdefender

Thanks for an awesome contribution!

**Pull request closed without merging**